### PR TITLE
catppuccin-bat: init at unstable-2022-11-10

### PR DESCRIPTION
**Pull request by khaneliman**
## Description of changes
Tested with home-manager bat theme config. https://github.com/nix-community/home-manager/blob/master/modules/programs/bat.nix#L48

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/data/themes/catppuccin-bat/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    runHook postInstall
+  '';
+
+  meta = {
+    description = "Soothing pastel theme for bat";
+    homepage = "https://github.com/catppuccin/bat";
+    license = lib.licenses.mit;
+    platforms = lib.platforms.linux;
+    maintainers = [ ];
```
Comment (aldoborrero) [inline pkgs/data/themes/catppuccin-bat/default.nix:38]: Why not add yourself to the maintainers list?  @khaneliman khaneliman

**Comment by khaneliman** [comment]
Fixed incorrect platform limitation.

**Comment by khaneliman** [comment]
@drupol @Artturin  got a few of this trying to make the catppuccin themes i use easier to reference from a config and figured it'd be nice to just put em in nixpkgs. 

- https://github.com/NixOS/nixpkgs/pull/259661
- https://github.com/NixOS/nixpkgs/pull/259664
- https://github.com/NixOS/nixpkgs/pull/259665
- https://github.com/NixOS/nixpkgs/pull/259831

File: pkgs/data/themes/catppuccin-bat/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+, variant ? "macchiato"
+}:
+let
+  pname = "catppuccin-bat";
+  validVariants = [ "latte" "frappe" "macchiato" "mocha" ];
+in
+lib.checkListOfEnum "${pname}: color variant" validVariants [ variant ]
+
+  stdenvNoCC.mkDerivation
+{
```
Comment (Artturin) [inline pkgs/data/themes/catppuccin-bat/default.nix:13]:
```suggestion
stdenvNoCC.mkDerivation {
```

Comment (khaneliman) [reply pkgs/data/themes/catppuccin-bat/default.nix]: Sorry, my nixpkgs-fmt kept making that change on me... i thought it preferred it this way too before.

**Comment by Artturin** [comment]
The maintainer commit can and should be in the init

**Comment by khaneliman** [comment]
Appreciate it, I made those same changes on the other PRs and will ping when the checks finish.

**Comment by Artturin** [comment]
Post-merge I got an idea, what if instead of separate packages you made one package which had an argument with the list of what themes should be in the package?

**Comment by Artturin** [comment]
> I had thought about something like that previously, too when it was getting tedious to create PRs and watching the themes folder fill up with these themes. Is there an example of a package similar to that style of combining multiple derivations into a single package output?

pseudo code

```
themes = [ something ];
validThemes = [ something something1 ]
srcSomething = ...
srcSomething1  = ...

checkListOfEnum validThemes themes

installPhase = lib.optionalString (lib.elem "something" themes) ''
  cp ${srcSomething}/Catppuccin-${variant}.tmTheme" $out/bat-Catpuccin-${variant}.tmTheme
 '' + lib.optionalString (lib.elem ...) ''
 '' ...
```

**Comment by khaneliman** [comment]
alright, that kinda looks like what i was reading. i'll try and throw that together in a new PR to simplify installing these themes

**Pull request merged**